server/backup_binlog_grpc_test.go

```diff
@@ -0,0 +1,257 @@
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/cybozu-go/moco"
+	"github.com/cybozu-go/moco-agent/metrics"
+	"github.com/cybozu-go/moco-agent/server/proto"
+	"github.com/cybozu-go/moco-agent/test_utils"
+	"github.com/cybozu-go/moco/accessor"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func testBackupBinloggRPC() {
+	var tmpDir string
+	var binlogDir string
+	var agent *Agent
+	var registry *prometheus.Registry
+	var sess *session.Session
+	var gsrv proto.BackupBinlogServiceServer
+
+	BeforeEach(func() {
+		var err error
+		tmpDir, err = ioutil.TempDir("", agentTestPrefix)
+		Expect(err).ShouldNot(HaveOccurred())
+		agent = New(test_utils.Host, clusterName, token, test_utils.MiscUserPassword, test_utils.CloneDonorUserPassword, replicationSourceSecretPath, tmpDir, replicaPort,
+			&accessor.MySQLAccessorConfig{
+				ConnMaxLifeTime:   30 * time.Minute,
+				ConnectionTimeout: 3 * time.Second,
+				ReadTimeout:       30 * time.Second,
+			},
+		)
+
+		By("creating MySQL and MinIO containers")
+		binlogDir, err = ioutil.TempDir("", binlogDirPrefix)
+		Expect(err).ShouldNot(HaveOccurred())
+		fmt.Println(binlogDir)
+		err = os.Chmod(binlogDir, 0777|os.ModeSetgid)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		test_utils.StopAndRemoveMySQLD(replicaHost)
+		err = test_utils.StartMySQLD(replicaHost, replicaPort, replicaServerID, binlogDir, binlogPrefix)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		test_utils.StopMinIO(agentTestPrefix + "minio")
+		err = test_utils.StartMinIO(agentTestPrefix+"minio", 9000)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("initializing MySQL replica")
+		err = test_utils.InitializeMySQL(replicaPort)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("initializing MinIO")
+		sess = session.Must(session.NewSession(&aws.Config{
+			Region:           aws.String("neco"),
+			Endpoint:         aws.String(fmt.Sprintf("%s:%d", "localhost", 9000)),
+			DisableSSL:       aws.Bool(true),
+			S3ForcePathStyle: aws.Bool(true),
+			Credentials:      credentials.NewStaticCredentials("minioadmin", "minioadmin", ""),
+		}))
+
+		Eventually(func() error {
+			return createBucket(sess, bucketName)
+		}, 10*time.Second).Should(Succeed())
+
+		By("setting environment variables for password")
+		os.Setenv(moco.RootPasswordEnvName, test_utils.RootUserPassword)
+
+		registry = prometheus.NewRegistry()
+		metrics.RegisterMetrics(registry)
+
+		gsrv = NewBackupBinlogService(agent)
+	})
+
+	AfterEach(func() {
+		By("deleting MySQL containers")
+		test_utils.StopAndRemoveMySQLD(donorHost)
+		test_utils.StopAndRemoveMySQLD(replicaHost)
+
+		By("deleting MinIO container")
+		test_utils.StopMinIO(agentTestPrefix + "minio")
+		os.RemoveAll(tmpDir)
+		os.RemoveAll(binlogDir)
+	})
+
+	It("should flush and backup binlog", func() {
+		By("calling /flush-backup-binlog API")
+		req := &proto.FlushAndBackupBinlogRequest{
+			Token:           token,
+			BackupId:        binlogPrefix,
```

Comment:
Could you define another variable like `backupID := "test-backup-id"`.
`binlogPrefix` is confusing.